### ISSUE TITLE
[Bug] Default constructor not found for type Xamarin.CommunityToolkit.UI.Views.Expander

### ISSUE THREAD

**Issue by Shunmugathangam** [xamarin/XamarinCommunityToolkit#1839]
Xamarin.Forms.Xaml.XamlParseException
  Message=Position 42:34. Default constructor not found for type Xamarin.CommunityToolkit.UI.Views.Expander
  Source=Xamarin.Forms.Xaml
  StackTrace:
  at Xamarin.Forms.Xaml.CreateValuesVisitor.Visit (Xamarin.Forms.Xaml.ElementNode node, Xamarin.Forms.Xaml.INode parentNode) [0x00363] in D:\a\1\s\Xamarin.Forms.Xaml\CreateValuesVisitor.cs:119 
  at Xamarin.Forms.Xaml.ElementNode.Accept (Xamarin.Forms.Xaml.IXamlNodeVisitor visitor, Xamarin.Forms.Xaml.INode parentNode) [0x000ac] in D:\a\1\s\Xamarin.Forms.Xaml\XamlNode.cs:158 
  at Xamarin.Forms.Xaml.ElementNode.Accept (Xamarin.Forms.Xaml.IXamlNodeVisitor visitor, Xamarin.Forms.Xaml.INode parentNode) [0x00079] in D:\a\1\s\Xamarin.Forms.Xaml\XamlNode.cs:154 
  at Xamarin.Forms.Xaml.ElementNode.Accept (Xamarin.Forms.Xaml.IXamlNodeVisitor visitor, Xamarin.Forms.Xaml.INode parentNode) [0x00079] in D:\a\1\s\Xamarin.Forms.Xaml\XamlNode.cs:154 
  at Xamarin.Forms.Xaml.ApplyPropertiesVisitor+<>c__DisplayClass44_0.<SetTemplate>b__0 () [0x000b7] in D:\a\1\s\Xamarin.Forms.Xaml\ApplyPropertiesVisitor.cs:785 
  at Xamarin.Forms.ElementTemplate.CreateContent () [0x00026] in D:\a\1\s\Xamarin.Forms.Core\ElementTemplate.cs:82 
  at Xamarin.Forms.Platform.Android.TemplatedItemViewHolder.Bind (System.Object itemBindingContext, Xamarin.Forms.ItemsView itemsView, System.Action`1[T] reportMeasure, System.Nullable`1[T] size) [0x00028] in D:\a\1\s\Xamarin.Forms.Platform.Android\CollectionView\TemplatedItemViewHolder.cs:61 
  at Xamarin.Forms.Platform.Android.ItemsViewAdapter`2[TItemsView,TItemsViewSource].BindTemplatedItemViewHolder (Xamarin.Forms.Platform.Android.TemplatedItemViewHolder templatedItemViewHolder, System.Object context) [0x00000] in D:\a\1\s\Xamarin.Forms.Platform.Android\CollectionView\ItemsViewAdapter.cs:140 
  at Xamarin.Forms.Platform.Android.StructuredItemsViewAdapter`2[TItemsView,TItemsViewSource].BindTemplatedItemViewHolder (Xamarin.Forms.Platform.Android.TemplatedItemViewHolder templatedItemViewHolder, System.Object context) [0x00038] in D:\a\1\s\Xamarin.Forms.Platform.Android\CollectionView\StructuredItemsViewAdapter.cs:103 
  at Xamarin.Forms.Platform.Android.ItemsViewAdapter`2[TItemsView,TItemsViewSource].OnBindViewHolder (AndroidX.RecyclerView.Widget.RecyclerView+ViewHolder holder, System.Int32 position) [0x00038] in D:\a\1\s\Xamarin.Forms.Platform.Android\CollectionView\ItemsViewAdapter.cs:76 
  at Xamarin.Forms.Platform.Android.StructuredItemsViewAdapter`2[TItemsView,TItemsViewSource].OnBindViewHolder (AndroidX.RecyclerView.Widget.RecyclerView+ViewHolder holder, System.Int32 position) [0x00056] in D:\a\1\s\Xamarin.Forms.Platform.Android\CollectionView\StructuredItemsViewAdapter.cs:92 
  at Xamarin.Forms.Platform.Android.SelectableItemsViewAdapter`2[TItemsView,TItemsSource].OnBindViewHolder (AndroidX.RecyclerView.Widget.RecyclerView+ViewHolder holder, System.Int32 position) [0x00000] in D:\a\1\s\Xamarin.Forms.Platform.Android\CollectionView\SelectableItemsViewAdapter.cs:22 
  at Xamarin.Forms.Platform.Android.GroupableItemsViewAdapter`2[TItemsView,TItemsViewSource].OnBindViewHolder (AndroidX.RecyclerView.Widget.RecyclerView+ViewHolder holder, System.Int32 position) [0x0004b] in D:\a\1\s\Xamarin.Forms.Platform.Android\CollectionView\GroupableItemsViewAdapter.cs:64 
  at AndroidX.RecyclerView.Widget.RecyclerView+Adapter.n_OnBindViewHolder_Landroidx_recyclerview_widget_RecyclerView_ViewHolder_I (System.IntPtr jnienv, System.IntPtr native__this, System.IntPtr native_holder, System.Int32 position) [0x00010] in D:\a\1\s\generated\androidx.recyclerview.recyclerview\obj\Release\monoandroid9.0\generated\src\AndroidX.RecyclerView.Widget.RecyclerView.cs:581 
  at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.157(intptr,intptr,intptr,int)

**Comment by VladislavAntonyuk** [comment]
could you please attach a small sample to reproduce the issue?

**Comment by Shunmugathangam** [comment]
Hi Antony

Note: Issue only exists in release build, but in debug build it works fine.

Step 1: 
Added **Xamarin.CommunityToolkit 2.0.1** reference in main, android and ios project.

Step2:
<?xml version="1.0" encoding="utf-8" ?>
<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             xmlns:xct="clr-namespace:Xamarin.CommunityToolkit.UI.Views;assembly=Xamarin.CommunityToolkit"
             x:Class="ATP.Reflection.Views.ReflectionListPage">
    <ContentPage.Content>
        <Grid RowSpacing="0"
              HorizontalOptions="FillAndExpand"
              VerticalOptions="FillAndExpand">
            <Grid.RowDefinitions>
                <RowDefinition Height="*" />
            </Grid.RowDefinitions>
            <xct:Expander>
                <xct:Expander.Header>
                    <Label Text="Xamarin Community Toolkit Expander"
                           FontAttributes="Bold"
                           FontSize="Medium" />
                </xct:Expander.Header>
                <Grid Padding="10">
                    <Grid.ColumnDefinitions>
                        <ColumnDefinition Width="Auto" />
                        <ColumnDefinition Width="Auto" />
                    </Grid.ColumnDefinitions>
                    <Image Source="https://devblogs.microsoft.com/xamarin/wp-content/uploads/sites/44/2021/01/admin-ajax.png"
                           Aspect="AspectFill"
                           HeightRequest="120"
                           WidthRequest="120" />
                    <Label Grid.Column="1"
                           Text="Xamarin Community Toolkit"
                           FontAttributes="Italic" />
                </Grid>
            </xct:Expander>
        </Grid>
    </ContentPage.Content>
</ContentPage>

Step3:
When try to open the screen, app crashes in the release mode (got the previous comment error details by unchecking the optimize code in android project and run the application)

Hope this helps, please let me know if you need more details.

Thanks.

**Comment by jtorvald** [comment]
@Shunmugathangam probably a linker issue when building for release mode.
Here you can see examples on how to keep the constructor while linking.
https://docs.microsoft.com/en-us/xamarin/cross-platform/deploy-test/linker

**Comment by jcslim89** [comment]
Had the same issue and managed to solve it by adding the following line to the Android project Linking options
![image](https://user-images.githubusercontent.com/48357926/218773703-a5049a66-f928-42f3-a182-fbaf3ea06302.png)
And for iOS for the MTOUCH arguments : 
--linkskip=Xamarin.CommunityToolkit
